src/fokker_planck_calculus.jl

```diff
@@ -0,0 +1,2279 @@
+    rpbd_exact::rosenbluth_potential_boundary_data,vpa,vperp;print_to_screen=true)
+    
+    error_buffer_vpa = Array{mk_float,1}(undef,vpa.n)
+    error_buffer_vperp_1 = Array{mk_float,1}(undef,vperp.n)
+    error_buffer_vperp_2 = Array{mk_float,1}(undef,vperp.n)
+    max_H_err = test_boundary_data(rpbd.H_data,rpbd_exact.H_data,"H",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_dHdvpa_err = test_boundary_data(rpbd.dHdvpa_data,rpbd_exact.dHdvpa_data,"dHdvpa",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_dHdvperp_err = test_boundary_data(rpbd.dHdvperp_data,rpbd_exact.dHdvperp_data,"dHdvperp",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_G_err = test_boundary_data(rpbd.G_data,rpbd_exact.G_data,"G",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_dGdvperp_err = test_boundary_data(rpbd.dGdvperp_data,rpbd_exact.dGdvperp_data,"dGdvperp",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_d2Gdvperp2_err = test_boundary_data(rpbd.d2Gdvperp2_data,rpbd_exact.d2Gdvperp2_data,"d2Gdvperp2",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_d2Gdvperpdvpa_err = test_boundary_data(rpbd.d2Gdvperpdvpa_data,rpbd_exact.d2Gdvperpdvpa_data,"d2Gdvperpdvpa",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+    max_d2Gdvpa2_err = test_boundary_data(rpbd.d2Gdvpa2_data,rpbd_exact.d2Gdvpa2_data,"d2Gdvpa2",vpa,vperp,error_buffer_vpa,error_buffer_vperp_1,error_buffer_vperp_2,print_to_screen)  
+
+    return max_H_err, max_dHdvpa_err, max_dHdvperp_err, max_G_err, max_dGdvperp_err, max_d2Gdvperp2_err, max_d2Gdvperpdvpa_err, max_d2Gdvpa2_err
+end
+
+function test_boundary_data(func,func_exact,func_name,vpa,vperp,buffer_vpa,buffer_vperp_1,buffer_vperp_2,print_to_screen)
+    nvpa = vpa.n
+    nvperp = vperp.n
+    for ivperp in 1:nvperp
+        buffer_vperp_1 = abs(func.lower_boundary_vpa[ivperp] - func_exact.lower_boundary_vpa[ivperp])
+        buffer_vperp_2 = abs(func.upper_boundary_vpa[ivperp] - func_exact.upper_boundary_vpa[ivperp])
+    end
+    for ivpa in 1:nvpa
+        buffer_vpa = abs(func.upper_boundary_vperp[ivpa] - func_exact.upper_boundary_vperp[ivpa])
+    end
+    max_lower_vpa_err = maximum(buffer_vperp_1)
+    max_upper_vpa_err = maximum(buffer_vperp_2)
+    max_upper_vperp_err = maximum(buffer_vpa)
+    if print_to_screen
+        println(string(func_name*" boundary data:"))
+        println("max(lower_vpa_err) = ",max_lower_vpa_err)
+        println("max(upper_vpa_err) = ",max_upper_vpa_err)
+        println("max(upper_vperp_err) = ",max_upper_vperp_err)
+    end
+    max_err = max(max_lower_vpa_err,max_upper_vpa_err,max_upper_vperp_err)
+    return max_err
+end

```

Comment:
I think there was an issue with wanting to use rosenbluth_potential_boundary_data to label the arguments of the function. I didn't want to make modules depend on one another as I wasn't sure which direction it should go in. Unfortunate, but at least the struct is in the same module so that it is easy to see how the function works.

Reply:
OK, makes sense.